docs/code/MlNetCookBook.md

```diff
@@ -0,0 +1,328 @@
+-0.85;-0.91;1.81;0.02;-0.78;-1.41;-1.09;-0.65;0.90;-0.37;-0.22;402.20
+```
+
+In the file above, the last column (12th) is label that we predict, and all the preceding ones are features.
+
+```c#
+// Create a new environment for ML.NET operations. It can be used for exception tracking and logging, 
+// as well as the source of randomness.
+var env = new LocalEnvironment();
+
+// Step one: read the data as an IDataView.
+// First, we define the reader: specify the data columns and where to find them in the text file.
+var reader = TextLoader.CreateReader(env, ctx => (
+        // We read the first 11 values as a single float vector.
+        FeatureVector: ctx.LoadFloat(0, 10),
+        // Separately, read the target variable.
+        Target: ctx.LoadFloat(11)
+    ),
+    // The data file has header.
+    hasHeader: true,
+    // Default separator is tab, but we need a semicolon.
+    separator: ';');
+
+
+// Now read the file (remember though, readers are lazy, so the actual reading will happen when the data is accessed).
+var trainData = reader.Read(new MultiFileSource(trainDataPath));
+
+// Step two: define the learning pipeline. 
+// We know that this is a regression task, so we create a regression context: it will give us the algorithms
+// we need, as well as the evaluation procedure.
+var regression = new RegressionContext(env);
+
+// We 'start' the pipeline with the output of the reader.
+var learningPipeline = reader.MakeNewEstimator()
+    // Now we can add any 'training steps' to it. In our case we want to 'normalize' the data (rescale to be
+    // between -1 and 1 for all examples), and then train the model.
+    .Append(r => (
+        // Retain the 'Target' column for evaluation purposes.
+        r.Target,
+        // We choose the SDCA regression trainer. Note that we normalize the 'FeatureVector' right here in
+        // the the same call.
+        Prediction: regression.Trainers.Sdca(label: r.Target, features: r.FeatureVector.Normalize())));
+
+// Step three. Train the pipeline.
+var model = learningPipeline.Fit(trainData);
+```
+
+## How do I verify the model quality?
+
+This is the first question that arises after you train the model: how good it actually is?
+For each of the machine learning tasks, there is a set of 'metrics' that can describe how good the model is: it could be log-loss or F1 score for classification, RMS or L1 loss for regression etc.
+
+You can use the corresponding 'context' of the task to evaluate the model.
+
+Assuming the example above was used to train the model, here's how you calculate the metrics.
+```c#
+// Read the test dataset.
+var testData = reader.Read(new MultiFileSource(testDataPath));
+// Calculate metrics of the model on the test data.
+// We are using the 'regression' context object here to perform evaluation.
+var metrics = regression.Evaluate(model.Transform(testData), label: r => r.Target, score: r => r.Prediction);
+```
+
+## How do I save and load the model?
```

Comment:
Is it worth covering how the loaded model is different from the model that was saved (pigsty vs. dynamic)?
Additionally, this describes operationalizing but not how to get a prediction. perhaps that's a different section.